### DISCUSSION TITLE
When debugging is enables, show the template/partial name in a comment in the source code

### DISCUSSION THREAD

**Discussion by jelle-burodeng** [ExpressionEngine/ExpressionEngine#2867] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
Sometimes it's hard to track what exact template/partial you are working with, especially with more complex sites that rely on layouts or ajax. You can only track so much in the debugging view

**Describe the solution you'd like**
I'd like to see a html comment on top and/or end of each block of rendered code that tells what template/partial/variable this is coming from like  or 

**Describe alternatives you've considered**
pulling my hair out each and every time

**Teachability, Documentation, Adoption, Migration Strategy**
If you can, explain how users will be able to use this and possibly write out a version the docs. Some of the best features are born from writing documentation first. Consider making a documentation pull request from a feature branch here: https://github.com/ExpressionEngine/ExpressionEngine-User-Guide.

- [ ] I am capable and would like to work on implementation of this feature if it is considered.

**Comment by intoeetive** [comment]
I've started working on this in https://github.com/ExpressionEngine/ExpressionEngine/pull/3700 but the realized this might be a bigger change than it initially seems.

We need to give people precize control whether they want / don't want this in specific templates (similarly to what we do for front-end editing) because it has potential to break stuff, depending on how template is built
